codex-rs/protocol/src/approvals.rs

```diff
@@ -53,6 +54,22 @@ pub struct ExecApprovalRequestEvent {
     pub parsed_cmd: Vec<ParsedCommand>,
 }
 
+#[derive(Debug, Clone, Deserialize, Serialize, JsonSchema, TS)]
+pub struct ElicitationRequestEvent {
+    pub server_name: String,
+    pub id: RequestId,
+    pub message: String,
+    // pub requested_schema: ElicitRequestParamsRequestedSchema,
```

Comment:
Maybe add a TODO or explain what else is required to support elicitations in full? (Admittedly it's a bit of a pain.)